app/models/token_url.rb

```diff
@@ -16,7 +16,7 @@ class TokenUrl < ActiveRecord::Base
   def self.valid?(token, notice)
     token_url = TokenUrl.find_by(token: token)
 
-    valid_token_url_and_notice?(token_url, notice) &&
+    [token_url.present?, notice.present?].all? &&
```

Comment:
The `[].all?` can encompass all 4 of these conditions.

Reply:
```
    [
      token_url.present?,
      notice.present?,
      valid_time?(token_url),
      token_url.notice == notice
    ].all?
```
You can do this here, when `token_url` is `nil` it will still call `valid_time?`.